Source/Manager/IdentityAPI/Models/UserDevice.swift

```diff
@@ -0,0 +1,59 @@
+//
+// Copyright 2011 - 2018 Schibsted Products & Technology AS.
+// Licensed under the terms of the MIT license. See LICENSE in the project root.
+//
+
+import Foundation
+
+/**
+ The user device data.
+
+ SeeAlso: https://techdocs.spid.no/types/device-fingerprint/
+ */
+public struct UserDevice: JSONParsable {
+    ///
+    private var hash: String?
```

Comment:
I think this should be separated into two classes for stricter typing and separation of the request and respose for `/devices`:
1. User device data sent to `/devices`, (maybe `UserDeviceFingerprint`): `deviceId` (required), `platform` (required), `applicationName ` (required), `applicationVersion` (optional)
1. Device fingerprint hash returned in the response from `/devices` (maybe `UserDeviceHash`): `hash` (required)

Reply:
Here I tried to be consistent with the existing approach in `UserProfile`:
```    
func updateUserProfile(
        userID: String,
        oauthToken: String,
        profile: UserProfile,
        completion: @escaping ((Result<UserProfile, ClientError>) -> Void)
)
```
Notice that the response/request share the same model.

Reply:
That endpoint (`/user/<userId>`) returns similar formatted data in the response as is sent in the request, basically echoing the stored user data for the given userId. So it makes more sense for that one, than the device fingerprint endpoint which returns completely different data in the response compared to the request.